### PR TITLE
Fixes #24043: API documentation for compliance by group

### PR DESCRIPTION
**Pull request by clarktsiory**
https://issues.rudder.io/issues/24043

Add public API documentation for these endpoints : 
  * `/api/compliance/groups/{groupId}` (global compliance) 
  * `/api/compliance/groups/{groupId}/target` (targeted compliance)

Also, there was a limitation in redocly to generate examples for nested objects, so we set the config parameter `generatedPayloadSamplesMaxDepth` to `100`. We also update redocly to `1.6.0`

### CODE REVIEW

**Comment by clarktsiory** [comment]
PR updated with a new commit

**Comment by clarktsiory** [comment]
PR updated with a new commit

File: webapp/sources/api-doc/paths/compliance/groups.yml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# SPDX-License-Identifier: CC-BY-SA-2.0
+# SPDX-FileCopyrightText: 2013-2024 Normation SAS
+get:
```
Comment (amousset) [inline webapp/sources/api-doc/paths/compliance/groups.yml]: ?

**Comment by clarktsiory** [comment]
PR updated with a new commit

**Comment by clarktsiory** [comment]
PR updated with a new commit

File: webapp/sources/api-doc/components/schemas/group-node-compliance-rules.yml [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+                type: array
+                items:
+                  type: object
+                  required:
+                    - value
+                    - reports
+                  properties:
+                    value:
+                      type: string
+                      example: tutu
```
[the remark this answers is not in the document]
Comment (clarktsiory) [reply webapp/sources/api-doc/components/schemas/group-node-compliance-rules.yml]: (I just copied it from the directives compliance documentation) 
https://github.com/Normation/rudder/pull/5332/commits/1fda62db140be51bbc7647c7c5a6eb6212aee65a

**Comment by amousset** [comment]
OK, squash merging this PR

**Pull request merged**